doc/decisions/DR018-Result-pager-design.md

```diff
@@ -0,0 +1,150 @@
+# DR018 Result pager design
+
+- **Status:** Decided
+- **Impact:** Medium
+- **Driver:** @foundrytom @feltech @elliotcmorris
+- **Approver:** @feltech @elliotcmorris
+- **Outcome:** Result set pagers will facilitate a general "moving
+  window" approach vs a simple one-shot queue.
+
+## Background
+
+Certain OpenAssetIO API methods can produce large sets of data.
+[DR016](DRDR016-One-to-many-result-paging.md) covers the evolution in
+thinking that lead us to expose this data through a lazy "result pager".
+
+These pagers are supplied by the relevant `Manager` methods. The caller
+then interacts directly with the pager to retrieve a sub-set of the
+data, _n_ results at a time.
+
+During the design of the pager API, it quickly became apparent that
+there were two possible ways think about the role of the pager.
+
+This DR describes these two high-level conceptual models, and which
+should be used when evolving the pager interface in the future.
+
+## Relevant data
+
+- [DR016](DRDR016-One-to-many-result-paging.md) opted for the pager
+  solution partly as it facilitates future extensions to data set
+  navigation with minimal disruption to the rest of the API.
+
+- By preference, OpenAssetIO opts for delegation over generalization -
+  to try and avoid the "lowest common denominator" trap awaiting
+  anyone who attempts to generalize the interactions between complex
+  systems.
+
+- Delegation also aims to encourage consistency across tools as
+  interactions with the manager are then presented via its native UI.
+
+## Options considered
+
+### Option 1 - "Simple queue"
+
+The pager is a marshal manning a simple queue. There is a large a number
+of results, and they are supplied to the caller a page at a time. Once
+`next` has been called, there is no way to retrieve the data
+for that page again.
+
+Its interface could look like this:
+
+```python
+class Pager:
+
+    def next() -> [Value]
+        """
+        Retrieves the next page of results.
+        """
+
+    def hasMore() -> Bool
+        """
+        Determines if calls to `next` will yield more results.
+        """
+```
+
+#### Pros
+
+- Simple, well understood data access pattern. Implementations are
+  guaranteed that the first page is always accessed.
+- Smaller surface are makes it easier to support across a variety of
+  back end implementations.
+- Discourages direct use of the API for complex workflows, encouraging
+  delegated UX, which can result in a more consistent user experience.
+
+#### Cons
+
+- Accessing data from anything other than the first page requires
+  traversal of all preceding pages.
+- Not all runtime environments support delegated UX and/or have critical
```

Comment:
[consider] This confused me a bit at first.  Perhaps worth pointing out in the description of this option that it is assumed UI delegation will pick up the slack for anything more complex.

Reply:
Added [here](https://github.com/OpenAssetIO/OpenAssetIO/pull/985/commits/9fdea11e410d40fd0e32bd21e29890b679c37009).